### PR TITLE
Create bug fix

### PR DESCRIPTION
**Pull request by mjudeikis**

### CODE REVIEW

File: pkg/env/dev.go [hunk 1]
```diff
@@ -256,6 +262,25 @@ func (d *dev) CreateARMResourceGroupRoleAssignment(ctx context.Context, fpAuthor
...
+	return wait.Poll(time.Second, time.Minute, func() (bool, error) {
+		// this should always error. Either 403 or 404
+		_, err := d.deployments.GetDeployment(ctx, oc.Properties.ResourceGroup, "dummy")
+		if detailedError, ok := err.(autorest.DetailedError); ok {
+			if requestError, ok := detailedError.Original.(azure.RequestError); ok &&
+				requestError.ServiceError != nil {
+				if requestError.ServiceError.Code == "AuthorizationFailed" {
+					return false, nil
+				}
+				if requestError.ServiceError.Code == "DeploymentNotFound" {
```
Comment (jim-minter) [inline pkg/env/dev.go]: not needed?

File: pkg/util/azureclient/mgmt/resources/deployments_addons.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ import (
 type DeploymentsClientAddons interface {
 	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) error
 	Wait(ctx context.Context, resourceGroupName string, deploymentName string) error
+	GetDeployment(ctx context.Context, resourceGroupName, deploymentName string) (resources.DeploymentExtended, error)
```
Comment (jim-minter) [inline pkg/util/azureclient/mgmt/resources/deployments_addons.go]: Add `Get(ctx context.Context, resourceGroupName, deploymentName string) (resources.DeploymentExtended, error)` to the DeploymentsClient interface instead

File: pkg/util/azureclient/mgmt/resources/deployments_addons.go [hunk 3]
```diff
@@ -41,3 +42,7 @@ func (c *deploymentsClient) Wait(ctx context.Context, resourceGroupName string,
 		return *deployment.Properties.ProvisioningState == "Succeeded", nil
 	})
 }
+
+func (c *deploymentsClient) GetDeployment(ctx context.Context, resourceGroupName, deploymentName string) (resources.DeploymentExtended, error) {
```
Comment (jim-minter) [inline pkg/util/azureclient/mgmt/resources/deployments_addons.go]: delete this

File: pkg/env/dev.go (hunk 1 above)
Comment (mjudeikis) [reply pkg/env/dev.go]: either we say - all errors, expect 403 is Fine - yes. Or we want to be explicit...

**Pull request merged**